halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/v2/KubernetesV2DeckService.java

```diff
@@ -70,6 +75,100 @@ public boolean runsOnJvm() {
     }
   }
 
+  @Override
+  public List<String> getInitContainers(AccountDeploymentDetails<KubernetesAccount> details) {
+    Plugins plugins = details.getDeploymentConfiguration().getPlugins();
+    if (plugins.isEnabled() && plugins.isDownloadingEnabled()) {
+      List<Map> initContainers =
+          details
+              .getDeploymentConfiguration()
+              .getDeploymentEnvironment()
+              .getInitContainers()
+              .getOrDefault(getServiceName(), new ArrayList<>());
+      initContainers.add(getPluginDownloadingContainer(plugins));
+      details
+          .getDeploymentConfiguration()
+          .getDeploymentEnvironment()
+          .getInitContainers()
+          .put(getServiceName(), initContainers);
+    }
+
+    return KubernetesV2Service.super.getInitContainers(details);
+  }
+
+  private Map<String, Object> getPluginDownloadingContainer(Plugins plugins) {
+    List<Manifest> pluginManifests =
+        plugins.getPlugins().stream()
+            .filter(p -> p.getEnabled())
+            .filter(p -> !p.getManifestLocation().isEmpty())
+            .map(p -> p.getManifest())
+            .filter(m -> m.getResources().containsKey(getCanonicalName()))
+            .distinct()
+            .collect(Collectors.toList());
+
+    Map<String, Object> pluginDownloadingContainer = new HashMap();
+    pluginDownloadingContainer.put("name", "plugin-downloader");
+    pluginDownloadingContainer.put("image", "busybox");
+    pluginDownloadingContainer.put("command", getCommand());
+    pluginDownloadingContainer.put("args", getPluginDownloadingArgs(pluginManifests));
+    pluginDownloadingContainer.put("volumeMounts", getVolumeMounts());
+    return pluginDownloadingContainer;
+  }
+
+  private List<String> getCommand() {
+    List<String> command = new ArrayList();
+    command.add("/bin/sh");
+    return command;
+  }
+
+  private List<String> getPluginDownloadingArgs(List<Manifest> manifests) {
+    StringBuilder sb = new StringBuilder("cd /opt/spinnaker/plugins/resources; ");
```

Comment:
instead of ending each shell command with a `;`, it might make more sense to use `&&` in case one of the commands fails, the rest of the commands will not run.